dspace-api/src/main/java/org/dspace/eperson/GroupServiceImpl.java

```diff
@@ -152,59 +157,80 @@ public boolean isDirectMember(Group group, EPerson ePerson) {
     }
 
     @Override
-    public boolean isMember(Group owningGroup, Group childGroup) {
-        return owningGroup.contains(childGroup);
+    public boolean isMember(Context context, Group owningGroup, Group childGroup) throws SQLException {
+        return group2GroupCacheDAO.findByParentAndChild(context, owningGroup, childGroup) != null;
```

Comment:
Please take a look into the javadoc in GroupService.isMember. This method should explicitly check only direct members. Using group2GroupCache is too much here. But we should add another method doing this.